src/overlays/actors/ovl_Obj_Chan/z_obj_chan.c

```diff
@@ -31,49 +39,401 @@ const ActorInit Obj_Chan_InitVars = {
+            temp_v0_2 = gSaveContext.weekEventReg[0x25];
+            if (!(temp_v0_2 & 0x10)) {
+                gSaveContext.weekEventReg[0x25] = temp_v0_2 | 0x10;
+                Actor_SpawnAsChildAndCutscene(&globalCtx->actorCtx, globalCtx, ACTOR_EN_MM, this->actor.world.pos.x,
+                                              this->actor.world.pos.y, this->actor.world.pos.z, 0, 0, 0, 0x8000,
+                                              this->actor.cutscene, this->actor.unk20, NULL);
+            }
+        }
+        Actor_MarkForDeath(&this->actor);
+    }
+}
+
+/**
+ * Plays some kind of effect when a pot is smashed
+ */
+void func_80BBA488(ObjChan* this, GlobalContext* globalCtx) {
+    short new_var = 0;
+    s32 phi_s0;
+    Vec3f spDC;
+    Vec3f spD0;
+    f32 temp_f0;
+    f32 new_var2;
+    s16 temp_s1 = 0;
+    s32 temp_s2;
+    f32 spAC;
+    f32 spA8 = new_var + 15.0f;
+    f32 spA4 = new_var + (spAC = 110.0f);
+
+    for (temp_s2 = 0, temp_s1 = 0; temp_s2 != 18; temp_s2++, temp_s1 += 0x4E20) {
+        float sin = Math_SinS(temp_s1);
+        float cos = Math_CosS(temp_s1);
+        spDC.x = sin * 8.0f;
+        spDC.y = Rand_ZeroOne() * 12.0f + 2.0f;
+        spDC.z = cos * 8.0f;
+        spD0.x = spDC.x * 0.23f;
+        spD0.y = Rand_ZeroOne() * 5.0f + 2.5f;
+        spD0.z = spDC.z * 0.23f;
+        temp_f0 = Rand_ZeroOne();
+        if (temp_f0 < 0.2f) {
+            phi_s0 = 0x60;
+        } else if (temp_f0 < 0.6f) {
+            phi_s0 = 0x40;
+        } else {
+            phi_s0 = 0x20;
+        }
+        new_var2 = spA4 * Rand_ZeroOne();
+        EffectSsKakera_Spawn(globalCtx, &spDC, &spD0, &this->actor.world.pos, -260, phi_s0, 20, 0, 0, spA8 + new_var2,
+                             0, 0, 50, -1, OBJECT_TSUBO, object_tsubo_DL_001960);
+    }
+    func_800BBFB0(globalCtx, &this->actor.world.pos, 30.0f, 2, 20, 50, true);
+    func_800BBFB0(globalCtx, &this->actor.world.pos, 30.0f, 2, 10, 80, true);
+}
+
+void ObjChan_Update(Actor* thisx, GlobalContext* globalCtx) {
+    ObjChan* this = THIS;
 
-extern ColliderCylinderInit D_80BBAB30;
-extern InitChainEntry D_80BBAB5C[];
+    this->actionFunc(this, globalCtx);
+    CollisionCheck_SetAC(globalCtx, &globalCtx->colChkCtx, &this->collider.base);
+}
 
-extern UNK_TYPE D_06000AF0;
-extern UNK_TYPE D_06001960;
-extern UNK_TYPE D_06002358;
+void ObjChan_Draw(Actor* thisx, GlobalContext* globalCtx) {
+    s32 pad;
+    ObjChan* this = THIS;
+    Gfx* dl_opa;
+    Gfx* dl_xlu;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Chan/ObjChan_Init.s")
+    OPEN_DISPS(globalCtx->state.gfxCtx);
+    Matrix_RotateY(this->rotation, MTXMODE_APPLY);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Chan/ObjChan_Destroy.s")
+    dl_opa = Gfx_CallSetupDL(POLY_OPA_DISP, 0x19);
+    gSPMatrix(dl_opa, Matrix_NewMtx(globalCtx->state.gfxCtx), G_MTX_LOAD);
+    gSPDisplayList(dl_opa + 1, object_obj_chan_DL_000AF0);
+    POLY_OPA_DISP = dl_opa + 2;
```

Comment:
```suggestion
    dl_opa = Gfx_CallSetupDL(POLY_OPA_DISP, 0x19);
    gSPMatrix(&dl_opa[0], Matrix_NewMtx(globalCtx->state.gfxCtx), G_MTX_LOAD);
    gSPDisplayList(&dl_opa[1], object_obj_chan_DL_000AF0);
    POLY_OPA_DISP = &dl_opa[2];
```
This is the preferred style for these kinds of scenarios. Applies elsewhere.